### PR TITLE
fix several cppcheck warnings in cLog and in gameobjects

### PR DESCRIPTION
**Pull request by pbomta**
Some unused variables an uninitialized things.
I refactored cGameObject to use cPoint, as we discussed when talking about cMap. I think that new classes should use a point and not separate x & y variables. But you may want to delete cGameObject for now? Let me know.

### CODE REVIEW

File: gameobjects/cGameObject.cpp [hunk 1]
```diff
@@ -1,17 +1,7 @@
-/*
- * cGameObject.cpp
- *
- *  Created on: 7-nov-2010
- *      Author: Stefan
- */
-
 #include "cGameObject.h"
```
Comment (stefanhendriks) [inline gameobjects/cGameObject.cpp:1]: afaik we don't use `cGameObject`. it was introduced with an idea but never pursued this. We could perhaps delete this and re-introduce when really needed later?

**Review by stefanhendriks — APPROVED** [review]
Lgtm overall. Nice improvements!

**Comment by stefanhendriks** [comment]
as for `I think that new classes should use a point and not separate x & y variables`

You are right. We should get rid of separate `x,y` as much as possible.

In fact, I'd want to get rid of `cell` as well if possible/handy. In the end I think structures, units, etc are all positioned on 'absolute coordinates' on a map. These absolute coordinates map with pixels btw. (which is fine I think for now)

File: gameobjects/cGameObject.cpp (hunk 1 above)
Comment (pbomta) [reply gameobjects/cGameObject.cpp:1]: Sounds good. So far, there is the split between an abstract structure and abstract unit, which works fine.

**Pull request merged**